### PR TITLE
#269 refactor/rename + fix behaviour in case of 2 bonds in intersection

### PR DESCRIPTION
**Pull request by ensemenova**

### CODE REVIEW

**Review by AndreiMazol — APPROVED** [review]

File: packages/ketcher-core/src/chemistry/entities/SGroup.ts [hunk 1]
```diff
@@ -187,32 +187,39 @@ export class SGroup {
...
+    return crossBonds
   }
 
-  static bracketPos(sg, mol, xbonds): void {
-    var atoms = sg.atoms
-    if (!xbonds || xbonds.length !== 2) {
-      sg.bracketDir = new Vec2(1, 0)
+  static bracketPos(sGroup, mol, crossBondsPerAtom: { [key: number]: Array<Bond> }): void {
+    var atoms = sGroup.atoms
+    const crossBonds = Array.prototype.concat.apply([], Object.values(crossBondsPerAtom))
```
Comment (TomasLight) [inline packages/ketcher-core/src/chemistry/entities/SGroup.ts:212]: please simplify 
`Array.prototype.concat.apply([], Object.values(crossBondsPerAtom))`
to 
`Object.values(crossBondsPerAtom).flat()`
or
`[].concat(...Object.values(crossBondsPerAtom))`

File: packages/ketcher-core/src/chemistry/entities/SGroup.ts [hunk 2]
```diff
@@ -230,23 +237,25 @@ export class SGroup {
           bbb = bbb === null ? new Box2Abs(p, p) : bbb!.include(p)
         })
       })
-      bb = bb === null ? bbb : Box2Abs.union(bb, bbb!)
+      braketBox = braketBox === null ? bbb : Box2Abs.union(braketBox, bbb!)
```
Comment (TomasLight) [inline packages/ketcher-core/src/chemistry/entities/SGroup.ts:240]: could you replace `braketBox === null` by `!braketBox` ?

File: packages/ketcher-core/src/chemistry/entities/SGroup.ts [hunk 3]
```diff
@@ -230,23 +237,25 @@ export class SGroup {
...
     })
     var vext = new Vec2(0.2, 0.4)
-    if (bb !== null) bb = bb!.extend(vext, vext)
-    sg.bracketBox = bb
+    if (braketBox !== null) braketBox = braketBox!.extend(vext, vext)
```
Comment (TomasLight) [inline packages/ketcher-core/src/chemistry/entities/SGroup.ts:243]: also, please replace null checking to boolean conversation `if (braketBox)`

File: packages/ketcher-core/src/chemistry/entities/SGroup.ts [hunk 4]
```diff
@@ -230,23 +237,25 @@ export class SGroup {
...
+    crossBondsPerAtom:{ [key: number]: Array<Bond> },
     atomSet: Pile<number>,
     bb,
     d,
     n
   ): Array<any> {
     var brackets: Array<any> = []
-    if (xbonds.length < 2) {
+    const crossBondsPerAtomValues = Object.values(crossBondsPerAtom)
+    const crossBonds = Array.prototype.concat.apply([], crossBondsPerAtomValues)
```
Comment (TomasLight) [inline packages/ketcher-core/src/chemistry/entities/SGroup.ts:257]: same as above

File: packages/ketcher-react/src/script/render/restruct/resgroup.js [hunk 5]
```diff
@@ -281,15 +279,16 @@ function drawAttachedDat(restruct, sgroup) {
...
   // eslint-disable-line max-statements
   var atoms = sg.atoms
 
-  if (!xbonds || xbonds.length !== 2) {
+  if (!crossBonds || crossBonds.length !== 2) {
     sg.bracketDir = new Vec2(1, 0)
   } else {
-    var p1 = mol.bonds.get(xbonds[0]).getCenter(mol)
-    var p2 = mol.bonds.get(xbonds[1]).getCenter(mol)
+    const crossBondsValues = Array.prototype.concat.apply([], Object.values(crossBonds))
```
Comment (TomasLight) [inline packages/ketcher-react/src/script/render/restruct/resgroup.js:289]: same as above

File: packages/ketcher-react/src/script/render/restruct/resgroup.js [hunk 6]
```diff
@@ -345,25 +344,28 @@ function getBracketParameters(mol, xbonds, atomSet, bb, d, render, id) {
   }
   var brackets = []
   var n = d.rotateSC(1, 0)
-  if (xbonds.length < 2) {
+
+  const crossBondsPerAtom = Object.values(crossBonds)
+  const crossBondsValues = Array.prototype.concat.apply([], crossBondsPerAtom)
```
Comment (TomasLight) [inline packages/ketcher-react/src/script/render/restruct/resgroup.js:349]: same

**Review by TomasLight — CHANGES_REQUESTED** [review]

File: packages/ketcher-core/src/chemistry/entities/SGroup.ts (hunk 2 above)
Comment (ensemenova) [reply packages/ketcher-core/src/chemistry/entities/SGroup.ts:240]: I'm not sure about such change

**Review by TomasLight — APPROVED** [review]

**Pull request merged**